vhost-device-sound/src/audio_backends/pipewire.rs

```diff
@@ -208,6 +208,15 @@ impl AudioBackend for PwBackend {
 
             let params = &stream_params[stream_id as usize].params;
 
+            if let Some(stream) = stream_hash.get(&stream_id) {
+                if let Err(err) = stream.disconnect() {
+                    log::error!("Stream {} disconnect {}", stream_id, err);
+                    return Err(Error::CouldNotDisconnectStream);
+                }
+                stream_hash.remove(&stream_id);
+                stream_listener.remove(&stream_id);
```

Comment:
Shouldn't the stream id be removed from those hashmaps even if it failed to disconnect?

Reply:
WDYM? The stream id will be removed regardless of whether the disconnection was successful or not.

Reply:
Not really per your code as you would return an error and the code I commented won't be executed